atlasdb-cli/src/test/java/com/palantir/atlasdb/cli/command/TestTimestampCommand.java

```diff
@@ -81,134 +81,204 @@ public TestAtlasDbServices connect(ServicesConfigModule servicesConfigModule) {
     }
 
     private SingleBackendCliTestRunner makeRunner(String... args) {
         return new InMemoryTestRunner(FetchTimestamp.class, args);
     }
 
+    private interface Verifier {
+        void verify(SingleBackendCliTestRunner runner, TimestampService tss, long immutableTs, long prePunch,
+                long postPunch, long lastFreshTs, boolean shouldTestImmutable) throws Exception;
+    }
+
     /*
      * Test that the various values produced by fetching either a fresh timestamp or
      * an immutable timestamp are correct relative to one another as well as the current
      * state of locks held by the lock service. isImmutable designates whether or not we
      * provided the argument to fetch the immutable timestamp rather than a fresh one.
      *
-     * Also make sure that the wallclock times associated with those timestamps are correct
+     * Also make sure that the wall clock times associated with those timestamps are correct
      * according to the current state of the _punch table.  We always provide the argument
      * to output this and thus always check it.
      *
      * Additionally, check that the formatting of commands that are writing the timestamps
-     * to a file for the purposes of scripting these commands is also correct.  isToFile
-     * designates whether or not we provided the argument to write to a file rather than just stdout
+     * to a file for the purposes of scripting these commands is also correct.
      */
     @Test
-    public void genericTest() throws Exception {
-        List<String> cliArgs = Lists.newArrayList("timestamp"); //group command
-        if (isToFile) {
-            cliArgs.add("-f");
-            cliArgs.add(TIMESTAMP_FILE_PATH);
+    public void testWithoutFile() throws Exception {
+        cliArgs.add("fetch");
+        if (isImmutable) {
+            cliArgs.add("-i");
         }
+        cliArgs.add("-d");
+
+        runAndVerifyCli((runner, tss, immutableTs, prePunch, postPunch, lastFreshTs, shouldTestImmutable) -> {
+            Scanner scanner = new Scanner(runner.run(true, false));
+            long timestamp = getTimestampFromStdout(scanner);
+            scanner.nextLine();
+            long wallClockTimestamp = getWallClockTimestamp(scanner);
+            scanner.close();
+            if (shouldTestImmutable && isImmutable) {
+                verifyImmutableTs(timestamp, immutableTs, prePunch, postPunch, lastFreshTs, tss.getFreshTimestamp(),
+                        wallClockTimestamp);
+            } else {
+                verifyFreshTs(timestamp, prePunch, postPunch, lastFreshTs, tss.getFreshTimestamp(), wallClockTimestamp);
+            }
+        });
+    }
+
+    @Test
+    public void testWithFile() throws Exception {
+        String inputFileString = "test.timestamp";
+        File inputFile = new File(inputFileString);
+        runAndVerifyCliForFile(inputFileString);
+        Files.delete(inputFile.toPath());
```

Comment:
Every Files.delete and permission restore should be done in an after block in case the test fails.